runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -111,6 +117,17 @@ local make_highlight_map = function(base_name)
   return result
 end
 
+local diagnostictag_highlight_map = {
+  [protocol.DiagnosticTag.Unnecessary] = {
+    group = "LspDiagnosticsUnnecessary",
```

Comment:
I think the existing groups are documented - so the new ones should get documented too.